### PR TITLE
K8SPSMDB-199 Init container

### PR DESCRIPTION
**Pull request by blez**

### CODE REVIEW

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| one-pod | passed |
| init-deploy | failed |
| upgrade | failed |
| storage | failed |
| upgrade-consistency | passed |
| limits | passed |
| monitoring | passed |
| self-healing | passed |
| demand-backup | passed |
| scaling | passed |
| monitoring-2-0 | failed |
| security-context | passed |
| scheduled-backup | passed |
| arbiter | passed |
| operator-self-healing | failed |
| service-per-pod | failed |
| liveness | passed |

File: e2e-tests/arbiter/compare/statefulset_arbiter-clusterip-rs0-arbiter-oc.yml [hunk 1]
```diff
@@ -120,6 +122,17 @@ spec:
           readOnly: true
         workingDir: /data/db
       dnsPolicy: ClusterFirst
+      initContainers:
+      - command:
+        - /init-entrypoint.sh
+        imagePullPolicy: IfNotPresent
```
Comment (delgod) [inline e2e-tests/arbiter/compare/statefulset_arbiter-clusterip-rs0-arbiter-oc.yml:128]:
```suggestion
        imagePullPolicy: Always
```

File: pkg/psmdb/init.go [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+	return corev1.Container{
+		VolumeMounts: []corev1.VolumeMount{
+			{
+				Name:      MongodDataVolClaimName,
+				MountPath: "/data/db",
+			},
+		},
+		Image:   initImageName,
+		Name:    "pxc-init",
+		Command: []string{"/init-entrypoint.sh"},
```
Comment (delgod) [inline pkg/psmdb/init.go:17]: init container should have the same resources requests/limits as the main container.
Could you please add?

File: e2e-tests/arbiter/compare/statefulset_arbiter-clusterip-rs0-arbiter-oc.yml (hunk 1 above)
Comment (blez) [reply e2e-tests/arbiter/compare/statefulset_arbiter-clusterip-rs0-arbiter-oc.yml]: https://github.com/percona/percona-server-mongodb-operator/pull/362/commits/b9076ea4547f7c038469a1ced001c0feaac3ee10

File: pkg/psmdb/init.go (hunk 2 above)
Comment (blez) [reply pkg/psmdb/init.go]: https://github.com/percona/percona-server-mongodb-operator/pull/362/commits/aab0abad6fa05278fd878a71c862cd133e5b1f33#diff-21cee571a261da42cae4cee0bf0952c6R78

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| storage | failed |
| init-deploy | failed |
| one-pod | failed |
| upgrade | passed |
| limits | failed |
| self-healing | failed |
| monitoring | passed |
| scaling | passed |
| upgrade-consistency | failed |
| security-context | passed |
| monitoring-2-0 | failed |
| operator-self-healing | failed |
| demand-backup | passed |
| arbiter | failed |
| service-per-pod | failed |
| scheduled-backup | failed |
| liveness | failed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| storage | failed |
| init-deploy | failed |
| limits | failed |
| self-healing | failed |
| scaling | failed |
| one-pod | failed |
| security-context | failed |
| monitoring | failed |
| monitoring-2-0 | failed |
| operator-self-healing | passed |
| arbiter | failed |
| service-per-pod | failed |
| liveness | failed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| storage | failed |
| init-deploy | failed |
| upgrade | failed |
| one-pod | failed |
| self-healing | failed |
| operator-self-healing | passed |
| monitoring | failed |
| upgrade-consistency | passed |
| limits | failed |
| scaling | passed |
| demand-backup | failed |
| scheduled-backup | failed |
| monitoring-2-0 | failed |
| arbiter | failed |
| security-context | passed |
| service-per-pod | failed |
| liveness | passed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | failed |
| one-pod | passed |
| limits | failed |
| upgrade | passed |
| scaling | failed |
| security-context | passed |
| monitoring | failed |
| upgrade-consistency | passed |
| monitoring-2-0 | passed |
| storage | failed |
| demand-backup | passed |
| self-healing | failed |
| arbiter | failed |
| scheduled-backup | passed |
| service-per-pod | failed |
| operator-self-healing | passed |
| liveness | passed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| one-pod | passed |
| monitoring | passed |
| storage | failed |
| upgrade | passed |
| upgrade-consistency | passed |
| demand-backup | passed |
| scheduled-backup | passed |
| init-deploy | passed |
| self-healing | passed |
| monitoring-2-0 | passed |
| arbiter | failed |
| limits | passed |
| scaling | failed |
| service-per-pod | passed |
| security-context | passed |
| operator-self-healing | passed |
| liveness | passed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| one-pod | passed |
| monitoring | passed |
| monitoring-2-0 | passed |
| arbiter | passed |
| init-deploy | passed |
| limits | passed |
| scaling | passed |
| upgrade | passed |
| upgrade-consistency | passed |
| demand-backup | passed |
| scheduled-backup | passed |
| storage | failed |
| security-context | passed |
| service-per-pod | passed |
| liveness | passed |
| self-healing | passed |
| operator-self-healing | passed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| storage | failed |
| one-pod | passed |
| init-deploy | passed |
| upgrade | passed |
| self-healing | passed |
| monitoring | passed |
| limits | passed |
| upgrade-consistency | passed |
| scaling | passed |
| demand-backup | failed |
| monitoring-2-0 | passed |
| security-context | passed |
| operator-self-healing | failed |
| scheduled-backup | passed |
| arbiter | passed |
| service-per-pod | passed |
| liveness | passed |

**Comment by JNKPercona** [comment]
PSMDB operator docker - perconalab/percona-server-mongodb-operator:PR-362-d7b52145

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| limits | passed |
| scaling | passed |
| security-context | passed |
| one-pod | passed |
| monitoring | passed |
| monitoring-2-0 | passed |
| arbiter | passed |
| storage | passed |
| service-per-pod | passed |
| liveness | passed |
| upgrade | passed |
| upgrade-consistency | passed |
| demand-backup | passed |
| self-healing | passed |
| operator-self-healing | passed |
| scheduled-backup | passed |

**Pull request merged**